internal/emulator/start.go

```diff
@@ -77,23 +77,22 @@ func ConfiguredServiceKey(
 		}
 	}
 
-	serviceAccount, _ := proj.EmulatorServiceAccount()
+	serviceAccount, err := proj.EmulatorServiceAccount()
+	if err != nil {
+		util.Exit(1, err.Error())
+	}
 
-	serviceKeyHex, ok := serviceAccount.DefaultKey().(*project.HexAccountKey)
-	if !ok {
+	privateKey, err := serviceAccount.Key().PrivateKey()
+	if err != nil {
 		util.Exit(1, "Only hexadecimal keys can be used as the emulator service account key.")
 	}
 
-	privateKey, err := crypto.DecodePrivateKeyHex(serviceKeyHex.SigAlgo(), serviceKeyHex.PrivateKeyHex())
+	err = serviceAccount.Key().Validate()
 	if err != nil {
-		util.Exitf(
-			1,
-			"Invalid private key in \"%s\" emulator configuration",
-			config.DefaultEmulatorConfigName,
-		)
+		util.Exit(1, err.Error())
```

Comment:
It'd be good to keep the original context of the error message to mention where this error originated from